src/SharedWeb/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -693,16 +695,33 @@ public static void AddWebAuthn(this IServiceCollection services, GlobalSettings
         this IServiceCollection services,
         GlobalSettings globalSettings)
     {
-        if (globalSettings.SelfHosted || string.IsNullOrEmpty(globalSettings.Redis.ConnectionString))
+        if (!string.IsNullOrEmpty(globalSettings.DistributedCache?.Redis?.ConnectionString))
+        {
+            services.AddStackExchangeRedisCache(options =>
+            {
+                options.Configuration = globalSettings.DistributedCache.Redis.ConnectionString;
+            });
+        }
+        else
         {
             services.AddDistributedMemoryCache();
-            return;
         }
 
-        services.AddStackExchangeRedisCache(options =>
+        if (!string.IsNullOrEmpty(globalSettings.DistributedCache?.Cosmos?.ConnectionString))
         {
-            options.Configuration = globalSettings.Redis.ConnectionString;
-        });
+            services.AddKeyedSingleton<IDistributedCache>("persistent", (s, _) =>
+                new CosmosCache(new CosmosCacheOptions
+                {
+                    DatabaseName = "cache",
+                    ContainerName = "default",
+                    CreateIfNotExists = false,
```

Comment:
❓ What's the impact of turning this on?

Reply:
It automagically creates it. I'd rather us be explicit about how it is created and configured.